src/Controller/CarController.php

```diff
@@ -294,16 +285,14 @@ public function delete(
         ]
     )]
 
-      public function expiringRegistration(CarRepository $carRepository): Response
-      {
-          $currentDate = new DateTimeImmutable();
-          $endOfNextMonth = $currentDate->modify('+30 days');
-                $cars = $carRepository->findByRegistrationExpiringUntil($endOfNextMonth);
-      
+      public function expiringRegistration(): Response
+    {
+        $cars = $this->carService->getCarsWithExpiringRegistration();
+
         return $this->render('car/expiring_registration.html.twig', [
             'cars' => $cars,
```

Comment:
instead of having $cars variable, put it directly here:
  'cars' =>  $this->carService->getCarsWithExpiringRegistration(); 
to save memory and  be clean

Reply:
Okey